File: testresources/csharp/benchmark/Domain/Direct/Violating/CallFromInnerClass.cs
using Technology.Direct.Dao;

namespace Domain.Direct.Violating
{
	public class CallFromInnerClass
	{
		// create an array
        private const int SIZE = 7;
        private int[] arrayOfInts = new int[SIZE];

		public CallFromInnerClass()
		{
			// fill the array with ascending integer values
			for (int i = 0; i < SIZE; i++)
			{
				arrayOfInts[i] = i;
			}
		}

        // inner class implements the Iterator pattern
		public class CallingInnerClass : Domain.Direct.Base
		{
			private readonly CallFromInnerClass _enclosing;
			private int next = 0;
			CallInstanceOuterClassDAO.CallInstanceInnerClassDAO innerClassDao;


			// start stepping through the array from the beginning
			public virtual bool HasNext()
			{
				return true;
			}

			public virtual int GetNext()
			{
				// record a value of an even index of the array
				int retValue = this._enclosing.arrayOfInts[this.next];
				//get the next even element
				this.next += 2;
				return retValue;
			}

			internal CallingInnerClass(CallFromInnerClass _enclosing)
			{
				this._enclosing = _enclosing;
                
			}
			
	    	public int CallMethodInstanceInnerClass() {
	    		int b;
	    		b = innerClassDao.getNext();

			int c = innerDao.getText();
	    		return c;
	    	}

		}
	}
}

File: testresources/java/accuracy/src/domain/direct/violating/CallFromInnerClass.java
package domain.direct.violating;

import technology.direct.dao.CallInstanceOuterClassDAO;

public class CallFromInnerClass {
    // create an array
    private final static int SIZE = 15;
    private int[] arrayOfInts = new int[SIZE];
    
    public CallFromInnerClass() {
        // fill the array with ascending integer values
        for (int i = 0; i < SIZE; i++) {
            arrayOfInts[i] = i;
        }
    }
    
    // inner class implements the Iterator pattern
    public class CallingInnerClass extends domain.direct.Base {
        // start stepping through the array from the beginning
        private int next = 0;

        private CallInstanceOuterClassDAO.CallInstanceInnerClassDAO innerDao; // Declaration of inner class, while only the outer class is imported

        public boolean hasNext() {
            return true ;
        }
        
        public int getNext() {
            // record a value of an even index of the array
            int retValue = arrayOfInts[next];
            //get the next even element
            next += 2;
            return retValue;
        }
        
    	public int CallMethodInstanceInnerClass() {
    		int b;
    		b = innerDao.getNext();
    		//AccessFromInnerClass
    		String t = innerDao.text;
    		return b;
    		
    	}

    }
}